### DISCUSSION TITLE
Problem Restore with constraints

### DISCUSSION THREAD

**Discussion by sebleje76** [mydumper/mydumper#1682] [Q&A]
Hi,

I'm having a problem restoring a schema because it's impossible to create tables with certain sql constraints. It gives me the error ‘errno: 150 “Foreign key constraint is incorrectly formed”.
Is there a way to create the tables and restore the constraints afterwards or create the table even if there is a problem with the constraints?

The version used is 0.16.5.

**Comment by davidducos** [comment]
Hi @sebleje76,  can you share the CREATE TABLE that is failing? as I need to review if it is a bug in myloader.
On myloader you have the option `--skip-constraints`.

**Reply by sebleje76** [reply]
hi

I tested with the skip-contraints command without success

**Reply by davidducos** [reply]
Hi @sebleje76, I need you to share information or a test case to be able to help you.

**Comment by sebleje76** [comment]
HI,
Attached is the command line used for restore
myloader --host localhost -u restore --password="xxxxxx’ -B middle_office -d /data_sav/Recup/20241128 -t 10 --skip-constraints --skip-indexes -v 3 -o -L /data_sav/Recup/restaure_middle_office_news.log

And here is the error message in the logs.

2024-11-29 07:44:36 [WARNING] - Connection 25228 - ERROR 1005: Can't create table `middle_office`.`article` (errno: 150 ‘Foreign key constraint is incorrectly formed’)

Package:
Nom          : mydumper
Version      : 0.16.5
Publication  : 1
Architecture : x86_64
Taille       : 19 M
Source       : mydumper-0.16.5-1.src.rpm

**Reply by davidducos** [reply]
can you share the CREATE TABLE that is failing?
you should be able to get it executing `SHOW CREATE TABLE middle_office.article`

**Comment by sebleje76** [comment]
[article.txt](https://github.com/user-attachments/files/17962447/article.txt)
I have just uploaded the sql creation file.
The article table is just an example, on the full restore I have 50 tables out of 300 that don't work.

**Reply by davidducos** [reply]
What version of database are you trying to import?

**Comment by sebleje76** [comment]
DB Source 10.3.8
Destination DB 10.5.22

**Comment by davidducos** [comment]
I was able to reproduce your issue. By default mydumper and myloader uses `/etc/mydumper.cnf` as the default file:
```
$ ./myloader --help | grep 'defaults-file'
  --defaults-file                       Use a specific defaults file. Default: /etc/mydumper.cnf
  --defaults-extra-file                 Use an additional defaults file. This is loaded after --defaults-file, replacing previous defined values
defaults-file                           = /etc/mydumper.cnf
```
which has important configuration like the session variables needed to import the backup. In your particular issue, the value of FOREIGN_KEY_CHECKS is not being set to 0. 
So, if you are using MacOS, you should check this link https://mydumper.github.io/mydumper/docs/html/installing.html#macos
If you are compiling from source and you have root privileges, execute `sudo make install` which is going to copy mydumper.cnf file to /etc and install mydumper and myloader in /usr/bin.
If you install it from a package, check if /etc/mydumper.cnf is there, if not, you can download it from the repo and use it with --defaults-file.

**Comment by sebleje76** [comment]
Thanks for your feedback, I'm going to test it tomorrow as the backup is running at the moment.
I have another question about mydumper updates.
If I want to upgrade mydumper (latest stable version) on an Oracle Linux 9 I get an error message below:
mydumper: error while loading shared libraries: libssl.so.10: cannot open shared object file: No such file or directory
Can you please help me

**Reply by davidducos** [reply]
you need to install compat-openssl11

**Comment by sebleje76** [comment]
HI,
When I run the command with the --default-file parameter, the program does not run (linux direct prompt). I tested the restore without the --skip-contraint and skip-index parameters and the restore was successful:
myloader --host localhost -u restore --password="xxxxxxxx’ -B middle_office -d /data_sav/Recup/20241128 -t 14 -v 3 -o -L /data_sav/Recup/restaure_middle_office_news_2.log

**Reply by davidducos** [reply]
what the log says?

**Comment by sebleje76** [comment]
Hi,
Hello,
I have installed the compact-ssl library and installed the 0.16.11-2 version of mydumper.
When I launch mydumper I get the error message 
ydumper: error while loading shared libraries: libssl.so.10: cannot open shared object file: No such file or directory

Best regard

**Reply by davidducos** [reply]
Seems that package libssl1.0.0 is missing.

**Reply by davidducos** [reply]
This is related to https://github.com/mydumper/mydumper/issues/1706
